cmd/influxd/launcher/cmd.go

```diff
@@ -74,17 +76,28 @@ func cmdRunE(ctx context.Context, o *InfluxdOpts) func() error {
 		ctx = signals.WithStandardSignals(ctx)
 
 		l := NewLauncher()
-		if err := l.run(ctx, o); err != nil {
+
+		// Set up logging to STDOUT.
+		logconf := &influxlogger.Config{
+			Format: "auto",
+			Level:  o.LogLevel,
+		}
+		log, err := logconf.New(os.Stdout)
+		if err != nil {
+			return err
+		}
+
+		if err := l.run(ctx, o, log); err != nil {
 			return err
 		}
 		<-ctx.Done()
 
 		// Attempt clean shutdown.
 		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
```

Comment:
i'm not sure if we can re-use a context this way.  you might need a new context value that isn't cancelled -- i don't think you can un-cancel a context chain.

Reply:
Will investigate, it'd be nice to squash a bonus bug as part of this

Reply:
This seems to match the pattern from `context.WithTimeout`'s godoc:
```go
// WithTimeout returns WithDeadline(parent, time.Now().Add(timeout)).
//
// Canceling this context releases resources associated with it, so code should
// call cancel as soon as the operations running in this Context complete:
//
// 	func slowOperationWithTimeout(ctx context.Context) (Result, error) {
// 		ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
// 		defer cancel()  // releases resources if slowOperation completes before timeout elapses
// 		return slowOperation(ctx)
// 	}
```
The following call to `l.Shutdown(ctx, log)` being the slow operation.